paddle/framework/lod_tensor.cc

```diff
@@ -152,6 +152,45 @@ void LoDTensor::ShrinkInLevel(size_t level, size_t elem_begin,
   ShareDataWith(Slice(begin, end));
 }
 
+void LoDTensor::SerializeToStream(
+    std::ostream &os, const platform::DeviceContext &dev_ctx) const {
+  Tensor::SerializeToStream(os, dev_ctx);
+  {  // serialize lod information
+     // uint64_t lod_level
+     // uint64_t lod_level_1 size in byte.
+     // int*     lod_level_1 data
+     // ...
+    auto lod = this->lod();
```

Comment:
lod information should never be Serialized, except for checkpoint(We will not do checkpoint at current stage). 
1. In save the model, lod information is useless.
2. In sending Tensor to ParameterServer, optimizer doesn't need the lod information.

Reply:
I have a PR implement the checkpoint, but our current design cannot fit it well. 
1. checkpoint must be Async Runnable. 
2. checkpoint need specify the synchronize parameters from GPUs, nodes.
3. Will checkpoint snapshot persistable object in Scope?  How about `Communicator` in NCCL?

Reply:
The conclusion is we will not touch that at the current stage. Leaving that in the future.

Reply:
I agree with `checkpoint'` will be left in the future, and we use the `Serialize` in saving the model and send/recv Op.

> lod information should never be Serialized

- It's not used in data parallelism, but maybe used in model parallelism.
- send/recv is a general Op, so I belive it will send/recv all kinds of variables which all information.

So mabye serialize the Lod information is a better way.